nodestream/project/project.py

```diff
@@ -213,6 +213,7 @@ def add_plugin_scope(self, name: str, plugin: PluginConfiguration):
         project_plugin_configuration = self.plugins_by_name.get(name)
 
         if project_plugin_configuration:
+            plugin.config = self.plugins_by_name.get(name).config
```

Comment:
Minimally, I think we can make this adjustment. 

```suggestion
            plugin.config = project_plugin_configuration.config
            plugin.update_pipeline_configurations(project_plugin_configuration)
```

However, I think I'd recommend moving this line down into `update_pipeline_configurations` and potentially renaming it so that `Project` does not have to be concerned with the inner data model of `PluginConfiguration`.

Reply:
whoops yeah wasn't done with the refactoring, did excatly that